linux/just_files/docker_functions.bsh

```diff
@@ -164,6 +165,41 @@ function is_readonly_docker_compose_long_volume()
   return 1
 }
 
+#**
+# .. function:: vsi::docker::is_docker_compose_volume_readonly
+#
+# :Arguments: * ``$1`` - A ``vsi_compose_volume_flags`` string to check (Not the whole array)
+# :Return Value: * ``0`` if it is readonly
+#                * ``1`` it if is writeable (according to docker flags only)
+#
+# Checks to see if a volume flags string contains the read only attribute.
+#
+# .. note::
+#
+#   It is still possible the underlying directory is not writeable, this will not check that. Used mostly for docker volumes, where this is not an issue.
+#
+# .. seealso::
+#
+#   vsi::docker::get_compose_volumes
+#     Parses docker compose volumes and generates ``vsi_compose_volume_flags``
+#**
+
+function vsi::docker::is_docker_compose_volume_readonly()
+{
+  local IFS=$'\n'
+  local flags=(${1}) # noquotes
+  local flag
+
+  for flag in ${flags[@]+"${flags[@]}"}; do
+    if [ "${flag}" == "read_only = true" ]; then
```

Comment:
is there any risk the spacing here can change, e.g., read_only=true

Reply:
I don't think so, that's the output from `yarp` which I control